src/ptc/util/jms.clj

```diff
@@ -75,6 +75,30 @@
          (map key->key)
          (into {}))))
 
+(defn update-cloud-path-keys
+  "If a file exists at the JMS cloud path, add the key to 'copy' otherwise
+  add the JMS key for the on-prem path to 'push'"
+  [wfl-key key-map workflow]
+  (let [[jms-cloud-key jms-on-prem-key] (get-in key-map [::push wfl-key])
+        on-prem-path (get workflow jms-on-prem-key)
+        cloud-path (get workflow jms-cloud-key)
+        uploaded? (try
+                 (misc/shell! "gsutil" "stat" cloud-path)
+                 (catch Exception _
+                   nil))]
+    (if (not uploaded?)
+      (misc/shell! "gsutil" "cp" on-prem-path cloud-path))
```

Comment:
If I understand right this will upload the on prem to the cloud path if it doesn't exist at the cloud path. I don't see an issue with that because then future invocations will see the file already being there and won't try to upload it again.